### ISSUE TITLE
Prevent Fluvio from creating multiple `minikubeCA` entries in /etc/hosts

### ISSUE THREAD

**Issue by nicholastmosher** [infinyon/fluvio#220]
Related to #90.

Currently, during the development setup process it is possible to get into a state where there are many lines of `minikubeCA` definition at the bottom of the `/etc/hosts` file. This causes some problems with tooling being able to connect to minikube and causes unreliability. We should fix this so that there is always at most one `minikubeCA`, and where our tools may update the line if necessary but not duplicate it.

I believe this is caused by `fluvio set-minikube-context` or `./dev-tools/minikube-mykube.sh`. On a related note, I believe the `sed` command in that script is not perfectly compatible with the MacOS version of `sed` (which may be the cause of this bug). `sed` gives this output on mac:

```
sed: 1: "/etc/hosts": extra characters at the end of h command
```

**Comment by sehz** [comment]
maybe use library like https://crates.io/crates/hostfile to parse and update to host file instead of using sed.

**Comment by nacardin** [comment]
Can this logic be built encapsulated as a rust crate so that it can be used in other projects?

**Comment by nicholastmosher** [comment]
I'm sure it can be, but I don't know if I want to get into making a full-fledged general-purpose crate for this. It seems like `hosts` files have [quite a long history](https://en.wikipedia.org/wiki/Hosts_(file)) and I don't really want to open the can of worms of making something more general than our specific use-case requires. I think it's something that could be fun in the future but I think one or two functions in one of our own crates could work just fine for this.

**Comment by nacardin** [comment]
I don't think there is anything specific about our use case. It is just syncing current `minikube ip` to "minikubeCA" in `/etc/hosts` and creating a new context in `.kube/config `. This is needed for cloud and in any other project that uses minikube and relies on rustls/webpki. If this cannot be shared, its going to result in duplicate code.

**Comment by nicholastmosher** [comment]
Working on this with infinyon/k8-api#12

**Comment by nicholastmosher** [comment]
Closed by infinyon/k8-api#12
